src/components/AddressHistory.js

```diff
@@ -73,27 +45,54 @@ class AddressHistory extends React.Component {
       return <p>This address does not have any transactions yet.</p>
     }
 
-    const getFirstHash = () => {
-      return this.props.transactions[0].tx_id;
+    const paginationLink = (page, query) => {
+        return this.props.pagination.setURLParameters({...query, page: page});
     }
 
-    const getLastHash = () => {
-      return this.props.transactions[this.props.transactions.length-1].tx_id;
-    }
+    const getPages = (page, total) => {
```

Comment:
We should have more comments explaining in details what this method is doing, I understand that you are selecting which pages will be shown in the pagination, however it's a bit confusing to me. A detailed explanation with examples would be great.

Reply:
Done.